lib/lightning_web/live/workflow_live/webhook_auth_method_form_component.ex

```diff
@@ -270,157 +270,163 @@ defmodule LightningWeb.WorkflowLive.WebhookAuthMethodFormComponent do
     """
   end
 
-  def render(%{action: :edit, show_2fa_options: true} = assigns) do
+  def render(assigns) do
     ~H"""
-    <div>
-      <.form
-        :let={f}
-        for={%{}}
-        action="#"
-        phx-submit="reauthenticate-user"
-        as={:user}
-        phx-target={@myself}
-        class="mt-2"
-        id="reauthentication-form"
-      >
-        <div class="space-y-4 ml-[24px] mr-[24px]">
-          <p class="font-normal text-sm whitespace-normal">
-            You're required to reauthenticate yourself before viewing the webhook
-            <%= if @webhook_auth_method.auth_type == :basic do %>
-              Password
-            <% else %>
-              API Key
-            <% end %>
-          </p>
-          <%= if @error_msg do %>
-            <div class="alert alert-danger" role="alert">
-              {@error_msg}
-            </div>
+    <div id={"write_webhook_auth_method_#{@id}"}>
+      <%= if @action == :edit and @show_2fa_options do %>
+        <.authenticate_user_form {assigns} />
+      <% else %>
+        <.webhook_auth_method_form {assigns} />
+      <% end %>
```

Comment:
Nice move! I think we should do the same, creating function components for this `case do` https://github.com/OpenFn/lightning/blob/9d8efdf36d94e558692e8e12a2b86775f650acac/lib/lightning_web/live/workflow_live/webhook_auth_method_form_component.ex#L356